web3/contract.py

```diff
@@ -753,6 +753,56 @@ def _encode_constructor_data(cls, args=None, kwargs=None):
         return deploy_data
 
 
+class ConciseContract:
+    '''
+    An alternative Contract Factory which invokes all methods as `call()`,
+    unless you add a keyword argument. The keyword argument assigns the prep method.
+
+    This call
+
+    > contract.withdraw(amount, transact={'from': eth.accounts[1], 'gas': 100000, ...})
+
+    is equivalent to this call in the classic contract:
+
+    > contract.transact({'from': eth.accounts[1], 'gas': 100000, ...}).withdraw(amount)
+
+    '''
+
+    def __init__(self, classic_contract):
+        self._classic_contract = classic_contract
+
+    @classmethod
+    def factory(cls, *args, **kwargs):
+        Prepped = Contract.factory(*args, **kwargs)
+        return compose(cls, Prepped)
+
+    def __getattr__(self, attr):
+        return ConciseMethod(self._classic_contract, attr)
+
+
+class ConciseMethod:
+
+    def __init__(self, contract, function):
+        self.__contract = contract
+        self.__function = function
+
+    def __call__(self, *args, **kwargs):
+        return self.__prepared_function(**kwargs)(*args)
+
+    def __prepared_function(self, **kwargs):
+        allowed_modifiers = ('call', 'estimateGas', 'transact')
```

Comment:
nitpick:  

* slightly more *correct* if this was a `set` rather than a tuple
* slightly more *correct* if this was defined once, maybe at the top level of this module.